src/unix/stream.c

```diff
@@ -839,11 +840,36 @@ static void uv__write(uv_stream_t* stream) {
 #endif
   } else {
     do {
+      /* On some platforms, notably OSX, the sum of iov_len must not
+       * overflow a 32-bit integer.
+       */
+      size_t old_iov_len = iov[iovcnt - 1].iov_len;
+
+      if (stream->write_queue_size > INT32_MAX) {
+        size_t total_bytes = 0;
+        int new_iov_cnt = 0;
+        int i;
+
+        for (i = 0; i < iovcnt; i++) {
+          new_iov_cnt++;
+          old_iov_len = iov[i].iov_len;
+          if ((total_bytes + iov[i].iov_len) >= INT32_MAX) {
+            iov[i].iov_len = INT32_MAX - (int32_t)total_bytes;
+            break;
+          }
+          total_bytes += iov[i].iov_len;
+        }
+
+        iovcnt = new_iov_cnt;
```

Comment:
This is logically the same as `iovcnt = i`, right?  I'd drop `new_iov_cnt`, one fewer variable to hold in your head while reading the code.